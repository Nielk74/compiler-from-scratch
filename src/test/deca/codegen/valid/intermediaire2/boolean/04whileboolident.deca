//Passe dans la boucle une fois, puis en sort à cause du x
{
int x = 5;
boolean y = true;
int a = 5;
boolean b = false;
while(y && (x < 10)){
  println("true && (5 < 10)");
  x = 10;
}
println("!(true && (10 < 10))");

//Ne passe pas dans la boucle à cause du y = false

while(b && (a < 10)){
  println("false && (5 < 10)");
  a = 10;
}
println("!(false && (5 < 10))");
}
