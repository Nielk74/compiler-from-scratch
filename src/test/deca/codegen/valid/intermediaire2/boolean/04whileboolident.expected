true && (5 < 10)
!(true && (10 < 10))
!(false && (5 < 10))
